libs/blocks/quiz/utils.js

```diff
@@ -317,24 +317,28 @@ export const findMatchForSelections = (results, selections) => {
     return defaultResult;
   }
   // Case 2 - when you have clauses grouped with parenthesis.
-  const compoundResults = results.find((destination) => {
-    if (destination.result.indexOf('(') !== -1 && destination.result.split('&').length === userSelectionLen) {
-      return destination;
+  const probableMatches = [];
+  results.forEach((rule) => {
+    if (rule.result.indexOf('(') !== -1 && rule.result.split('&').length === userSelectionLen) {
+      probableMatches.push(rule);
     }
   });
```

Comment:
nit
```suggestion
const probableMatches = results.reduce((match, rule) => {
  if (rule.result.includes('(') && rule.result.split('&').length === userSelectionLen) {
    match.push(rule);
  }
  return match;
}, []);
```

Reply:
Looks much better. Thanks for the feedback. Updated.